src/status_im2/contexts/wallet/events.cljs

```diff
@@ -72,35 +75,54 @@
 
 (rf/reg-event-fx
  :wallet/save-account
- (fn [_ [{:keys [account callback]}]]
+ (fn [_ [{:keys [account on-success]}]]
    {:fx [[:json-rpc/call
           [{:method     "accounts_saveAccount"
             :params     [(data-store/<-account account)]
             :on-success (fn []
                           (rf/dispatch [:wallet/get-accounts])
-                          (when (fn? callback)
-                            (callback)))
+                          (when (fn? on-success)
+                            (on-success)))
             :on-error   #(log/info "failed to save account "
                                    {:error %
                                     :event :wallet/save-account})}]]]}))
 
-(rf/reg-event-fx :wallet/get-wallet-token
+(rf/reg-event-fx
+ :wallet/get-wallet-token
  (fn [{:keys [db]}]
-   (let [addresses (map :address (vals (get-in db [:wallet :accounts])))]
+   (let [addresses (->> (get-in db [:wallet :accounts])
+                        vals
+                        (map :address))]
      {:fx [[:json-rpc/call
             [{:method     "wallet_getWalletToken"
               :params     [addresses]
-              :on-success [:wallet/get-wallet-token-success]
+              :on-success [:wallet/store-wallet-token]
               :on-error   #(log/info "failed to get wallet token "
                                      {:error  %
                                       :event  :wallet/get-wallet-token
                                       :params addresses})}]]]})))
 
-(rf/reg-event-fx :wallet/get-wallet-token-success
- (fn [{:keys [db]} [tokens]]
-   {:db (assoc db
-               :wallet/tokens          tokens
-               :wallet/tokens-loading? false)}))
+(defn- fix-chain-id-keys
+  [token]
+  (update token :balances-per-chain update-keys (comp utils.number/parse-int name)))
+
+(rf/reg-event-fx
+ :wallet/store-wallet-token
+ (fn [{:keys [db]} [raw-tokens-data]]
+   (let [tokens     (-> raw-tokens-data
+                        (update-keys name)
+                        (update-vals #(cske/transform-keys csk/->kebab-case %))
+                        (update-vals #(mapv fix-chain-id-keys %)))
+         add-tokens (fn [stored-accounts tokens-per-account]
+                      (reduce-kv (fn [accounts address tokens-data]
+                                   (if (accounts address)
+                                     (update accounts address assoc :tokens tokens-data)
+                                     accounts))
+                                 stored-accounts
+                                 tokens-per-account))]
+     {:db (-> db
+              (update-in [:wallet :accounts] add-tokens tokens)
+              (assoc-in [:wallet :ui :tokens-loading?] false))})))
```

Comment:
tokens data are now inside accounts and the `tokens-loading?` is in a separate branch